### DISCUSSION TITLE
Retina Api server latency

### DISCUSSION THREAD

**Discussion by kevin-intri-form3** [microsoft/retina#503] [Q&A] [answered]
Hi everyone.

We are exploring this amazing project and we are interested in latency metrics, [`adv_node_apiserver_latency `](https://retina.sh/docs/metrics/advanced#plugin-packetparser-linux) specifically.
Something we don't know is what means API server 🤔 
We though it could be the k8s api server, but we are not sure about that because we found some configs around it and the same port value used for retina is used as api port.

Could you please help us to understand better the context of the metric flow?

Thanks!

**Comment by SRodi** [comment]
@alexcastilio

**Comment by anubhabMajumdar** [comment, marked as answer]
Hi @kevin-intri-form3 , thanks for your interest in the project.
You are correct, `apiserver` refers to the Kubernetes API server. The `adv_node_apiserver_latency` refers to TCP latency between a node and K8s `apiserver`. There's three different metrics that provides 3 important latency signal:
- adv_node_apiserver_latency
- adv_node_apiserver_no_response
- adv_node_apiserver_tcp_handshake_latency

You can read more about this here - https://retina.sh/docs/Metrics/modes/advanced/#plugin-packetparser-linux .

Let me know if this helps! Cheers!
